spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -1582,19 +1582,22 @@ def paste(self):
         text has 'CRLF' EOL chars
         """
         clipboard = QApplication.clipboard()
-        # This is here to prevent pasting mime data in the Editor.
-        # See issue 8566 for the details.
+        text = to_text_string(clipboard.text())
+        # This is here to make copied files/folders to be pasted as path.
+        # See issue 8566 and PR: 8576 for the details.
         if clipboard.mimeData().hasUrls():
-            return
-        else:
-            text = to_text_string(clipboard.text())
-            if len(text.splitlines()) > 1:
-                eol_chars = self.get_line_separator()
-                text = eol_chars.join((text + eol_chars).splitlines())
-                clipboard.setText(text)
-            # Standard paste
-            TextEditBaseWidget.paste(self)
-            self.document_did_change(text)
+            urls = clipboard.mimeData().urls()
+            if len(urls) > 1:
+                text = ''.join('"' + url.toLocalFile().replace(osp.os.sep, '/')
+                               + '",' + '\n' for url in urls)
+            else:
+                text = urls[0].toLocalFile() + '\n'
+        eol_chars = self.get_line_separator()
+        text = eol_chars.join((text + eol_chars).splitlines())
+        clipboard.setText(text)

```

Comment:
The issue here is we're overwriting the content of the global system clipboard, which is not the user-expectation when pasting. Can you either save and restore the previous clipboard contents once done pasting here, or add a ``text`` parameter to ``TextEditBaseWidget`` (defaulting to ``None`` with the current behavior, of course) that allows passing text to paste directly to that function?